src/trace-api.ts

```diff
@@ -241,6 +242,17 @@ export class StackdriverTracer implements Tracer {
             rootSpan.span.name}] was already closed.`);
         return UNCORRELATED_CHILD_SPAN;
       }
+      if (rootSpan.trace.spans.length === this.config!.spansPerTraceSoftLimit) {
```

Comment:
is this the only code path where something is added to `rootSpan.trace.spans`?
just curious if the === will always hit. I didn't quite finish tracing the code path

Reply:
Good catch -- you're right that this is not always the case, as `rootSpan` has an API to directly create child spans, which circumvents this code. However, no automatic tracing plugin uses the API (because no plugin creates both root and child spans from a single module in the same continuation). So unless the user is specifically using the custom span API, in which case they are circumventing the use of context propagation entirely, they will see this message if they rack up enough child spans.

I will add this as a comment.

Reply:
BTW -- re: your approval comment, in the meantime, I also added tests.